types/context.go

```diff
@@ -268,11 +268,18 @@ func (c Context) TransientStore(key storetypes.StoreKey) KVStore {
 
 // CacheContext returns a new Context with the multi-store cached and a new
 // EventManager. The cached context is written to the context when writeCache
-// is called.
+// is called. Note, events are automatically emitted on the parent context's
+// EventManager when the caller executes the write.
 func (c Context) CacheContext() (cc Context, writeCache func()) {
 	cms := c.MultiStore().CacheMultiStore()
 	cc = c.WithMultiStore(cms).WithEventManager(NewEventManager())
-	return cc, cms.Write
+
+	writeCache = func() {
```

Comment:
this is the breaking part, should we spin it out into its own function and document it?

Reply:
I don't see how this is API breaking personally?

Reply:
Because it is a behavior change. If we are fine with that, let me revert the addition and merge as it :)

Reply:
I see. This isn't an API break (changelog states that it is). It does introduce _new_ or _additive_ behavior, so I dont think its breaking. But I wont object to this PR either 👍